Schlosskirsch/Objects/HealthObject.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Schlosskirsch.Objects
+{
+    public abstract class HealthObject : GameObject
```

Comment:
Ich würde Enemy auch davon ableiten lassen, da die sich eigentlich genauso wie der Player verhalten sollten, nur ohne Healthbar bzw. bei stärkeren eventuell mit. Generell fände ich es aber besser, daraus eine Komponente zu machen, weil das für mich eher eine "has a" und weniger eine "is a" Beziehung ist.

Reply:
Die Ableitungsstruktur ist meiner Meinung nach sowieso nicht die beste... Das laden der Texturen sollte innerhalb der Klasse passieren... Enemies sollten auch eine Weapon haben... die Kugeln sollten eine maximale Reichweite haben... usw.
Ich würde durchaus von HealthObject für Enemies ableiten... sehe da keinen Nachteil... aber es ist ja dein Spiel ;)